src/iio_wrapper.cpp

```diff
@@ -78,8 +81,8 @@ IIOWrapper::IIOWrapper()
             }
         }
 
-        // enable first 7 channels
-        for (int i = 0; i <= 6; ++i)
+        // enable first all channels
+        for (int i = 0; i < m_channel_count; ++i)
```

Comment:
this should be the only change done here...just remove the version in which you use the 7 channels